### PR TITLE
[CC-26337] Version bump for Jackson Databind package

### PR DESCRIPTION
**Pull request by sp-gupta**
## Problem
https://confluentinc.atlassian.net/browse/CC-26337

Few issues with updating `common` package:

1. Usually in `common` repo, artifacts are created as soon as a commit is merged to a feature branch. But 6.0.x is not migrated to Semaphore. Hence, no artifacts are generated for this branch. Also, there is no provision to run connect-releases job on this repo. Hence migrated the common package to 6.1.x branch.

2. The converters (Avro, Json etc) use `confluent.version` parameter which translates to the common package version (6.1.16-32 in this case). Upon checking the maven packages available for [Kafka Avro converter](https://mvnrepository.com/artifact/io.confluent/kafka-connect-avro-converter), it is available till 6.1.15 
So we would need to pin the converter artifacts to 6.1.15 version. 

## Solution
We can pin the jackson version to 2.16.0 in the pom. Created a new ticket [here](https://confluentinc.atlassian.net/browse/CC-26830) to bump to new common version and remove jackson changes from this pom once the CP release is done at the end of the quarter.

##### Does this solution apply anywhere else?
- [ ] yes
- [ ] no

##### Testing done:
- [ ] Unit tests
- [ ] Integration tests
- [ ] System tests
- [ ] Manual tests

Refer to the comment: https://confluentinc.atlassian.net/browse/CC-26337?focusedCommentId=2591491

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -52,6 +52,7 @@
         <confluent.maven.repo>http://packages.confluent.io/maven/</confluent.maven.repo>
         <commons.codec.version>1.15</commons.codec.version>
         <snakeyaml.version>1.28</snakeyaml.version>
+        <jackson.version>2.16.0</jackson.version>
```
Comment (ngupta0205) [inline pom.xml:55]: As discussed over the call, let's make this change in the parent pom only. That would facilitate others to not pin their individual versions to fix this CVE, rather just upgrade the parent pom version.

File: src/test/java/io/confluent/connect/elasticsearch/MappingTest.java [hunk 2]
```diff
@@ -38,6 +38,7 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+@SuppressWarnings("deprecation")
```
Comment (ngupta0205) [inline src/test/java/io/confluent/connect/elasticsearch/MappingTest.java:41]: why is this needed?

**Review by ngupta0205 — APPROVED** [review]
Left a minor comment. LGTM otherwise.

File: src/test/java/io/confluent/connect/elasticsearch/MappingTest.java (hunk 2 above)
Comment (sp-gupta) [reply src/test/java/io/confluent/connect/elasticsearch/MappingTest.java:41]: ObjectNode is getting deprecated so the build fails with some warnings.

**Review by aniketshrimal — APPROVED** [review]
LGTM @sp-gupta! Lets create a ticket to remove this pinning once common version is released with the change.

**Review by ngupta0205 — APPROVED** [review]
